File: backend-api/internal/models/archive.go
package models

type CountTotal struct {
	Total int `json:"total"`
}

type ArchiveCounts struct {
	Cards CountTotal `json:"cards"`

	Equipments CountTotal `json:"equipments"`

	Headwears CountTotal `json:"headwears"`

	Monsters CountTotal `json:"monsters"`

	Mounts CountTotal `json:"mounts"`

	Pets CountTotal `json:"pets"`

	Skills CountTotal `json:"skills"`

	Buffs CountTotal `json:"buffs"`

	Formulas CountTotal `json:"formulas"`
}

File: archive-romhandbook-ui/lib/types/Archive.ts
export type CountTotal = {

    total: number
}

export type ArchiveCounts = {

    cards: CountTotal

    equipments: CountTotal

    headwears: CountTotal

    monsters: CountTotal

    mounts: CountTotal

    pets: CountTotal

    skills: CountTotal

    buffs: CountTotal

    formulas: CountTotal
}

export type ApiResponse<T> = {

    success: boolean

    data: T

    meta: unknown
}